lib/components/os/resinos.js

```diff
@@ -71,3 +67,28 @@ exports.injectNetworkConfiguration = (image, configuration) => {
     path: '/system-connections/resin-wifi'
   }, wifiConfiguration.join('\n'))
 }
+
+module.exports = class ResinOS {
+  constructor (image, fetch, configuration) {
+    this.image = image
+    this.fetch = fetch
+    this.configuration = configuration
+  }
+
+  async runFetch () {
+    console.log(`Downloading device type OS into ${this.image}`)
+    await this.fetch(this.image)
```

Comment:
This also looks like a very strange pattern. I'm starting to think these classes should take as a constructor argument the path where the image already lives (was fetched before instantiating the class), but then the classes may provide a `.fetch()` static function that encapsulates this logic for the client to use